scripts/workarounds/boot-order/metal-lib.sh

```diff
@@ -0,0 +1,510 @@
+    # remove trailing commas in case 'cat /tmp/rbbs*' is empty
+    boot_order=$(echo ${efibootmgr_prefix}${boot_order_bbs},${boot_order_rbbs} | sed 's/,*$//g')
+    echo enforcing boot order $(cat /tmp/bbs*) && efibootmgr -o $boot_order | grep -i bootorder
+    echo activating boot entries && cat /tmp/bbs* | awk '!x[$0]++' | sed 's/^Boot//g' | tr -d '*' | awk '{print $1}' | xargs -r -i efibootmgr -b {} -a
+}
+
+# uses /tmp/rbbs99
+function efi_specials {
+    # TODO: If Marvell; then ensure PXE retries only once per NIC.
+    echo 'removing Shasta V1.3 items' && efibootmgr | grep -iP '(crayinstall|sles-secureboot)' | tee /tmp/sbbs
+}
+
+function setup_uefi_bootorder() {
+cat << EOM
+Configuring UEFI boot-order...
+these use the same commands from the manual page:
+    https://github.com/Cray-HPE/docs-csm/blob/main/background/ncn_boot_workflow.md#setting-order
+EOM
+    echo scanning vendor ... && vendor=$(ipmitool fru | grep -i 'board mfg' | tail -n 1 | cut -d ':' -f2 | tr -d ' ')
+    hostname=${hostname:-$(hostname)}
+    # Add vendors here; add like-vendors on the same case statement.
+    # "like-vendors" means their efibootmgr outboot matches
+
+    # formatting:
+    # if another vendor is identical then it should live with another.
+    # vendors may have differing hostnames, depending where this script runs
+    # vendor)
+    #   hostname_prefix_1)
+    #     file1)
+    #     fileN)
+    #   hostname_prefix_2)
+    #     file1)
+    #     fileN)
+    #   hostname_prefix_N)
+    #     file1)
+    #     fileN)
+    #   error)
+    #   remove_file_1
+    #   remove_file_N
+    # done
+    case $vendor in
+        *GIGABYTE*)
+            # Removal file(s) ...
+            efibootmgr | grep -ivP '(pxe ipv?4.*)' | grep -iP '(adapter|connection|nvme|sata)' | tee /tmp/rbbs1
+            efibootmgr | grep -iP '(pxe ipv?4.*)' | grep -i connection | tee /tmp/rbbs2
+            efibootmgr_prefix=''
+            efi_trim
+            efi_specials
+            efi_remove
+            case $hostname in
+                ncn-m*)
+                    efibootmgr | grep -iP '(pxe ipv?(4|6).*adapter)' | tee /tmp/bbs1
+                    efibootmgr | grep -i cray | tee /tmp/bbs2
+                    efibootmgr | grep 'UEFI OS' | tee /tmp/bbs3
+                    ;;
+                ncn-s*)
+                    efibootmgr | grep -iP '(pxe ipv?(4|6).*adapter)' | tee /tmp/bbs1
+                    efibootmgr | grep -i cray | tee /tmp/bbs2
+                    efibootmgr | grep 'UEFI OS' | tee /tmp/bbs3
+                    ;;
+                ncn-w*)
+                    efibootmgr | grep -iP '(pxe ipv?(4|6).*adapter)' | tee /tmp/bbs1
+                    efibootmgr | grep -i cray | tee /tmp/bbs2
+                    efibootmgr | grep 'UEFI OS' | tee /tmp/bbs3
+                    ;;
+                *)
+                    efi_fail_host
+                    ;;
+            esac
+            ;;
+        *Marvell*|HP|HPE)
+            # Removal file(s) ...
```

Comment:
```suggestion
            # Removal files ...
```

Reply:
It pains me but I'm going to omit amending the metal-lib.sh script with changes to keep the diff to the original script in 1.3/1.4 the same.